pkg/connector/handlematrix.go

```diff
@@ -391,3 +400,133 @@ func (wa *WhatsAppClient) HandleMatrixMembership(ctx context.Context, msg *bridg
 
 	return true, nil
 }
+
+func (wa *WhatsAppClient) HandleMatrixRoomName(ctx context.Context, msg *bridgev2.MatrixRoomName) (bool, error) {
+	portalJID, err := waid.ParsePortalID(msg.Portal.ID)
+	if err != nil {
+		return false, err
+	}
+
+	if msg.Portal.RoomType == database.RoomTypeDM {
+		return false, fmt.Errorf("cannot set room name for DM")
+	}
+
+	err = wa.Client.SetGroupName(portalJID, msg.Content.Name)
+	if err != nil {
+		return false, err
+	}
+
+	msg.Portal.Name = msg.Content.Name
+	msg.Portal.NameSet = true
+
+	return true, nil
+}
+
+func (wa *WhatsAppClient) HandleMatrixRoomTopic(ctx context.Context, msg *bridgev2.MatrixRoomTopic) (bool, error) {
+	portalJID, err := waid.ParsePortalID(msg.Portal.ID)
+	if err != nil {
+		return false, err
+	}
+
+	if msg.Portal.RoomType == database.RoomTypeDM {
+		return false, fmt.Errorf("cannot set room topic for DM")
+	}
+
+	err = wa.Client.SetGroupTopic(portalJID, "", "", msg.Content.Topic)
+	if err != nil {
+		return false, err
+	}
+
+	msg.Portal.Topic = msg.Content.Topic
+	msg.Portal.TopicSet = true
+
+	return true, nil
+}
+
+func (wa *WhatsAppClient) HandleMatrixRoomAvatar(ctx context.Context, msg *bridgev2.MatrixRoomAvatar) (bool, error) {
+	portalJID, err := waid.ParsePortalID(msg.Portal.ID)
+	if err != nil {
+		return false, err
+	}
+
+	if msg.Portal.RoomType == database.RoomTypeDM {
+		return false, fmt.Errorf("cannot set room avatar for DM")
+	}
+
+	var data []byte
+	if msg.Content.URL == "" {
+		data = nil
+	} else {
+		data, err = msg.Portal.Bridge.Bot.DownloadMedia(ctx, msg.Content.URL, msg.Content.MSC3414File)
+		if err != nil {
+			return false, fmt.Errorf("failed to download avatar: %w", err)
+		}
+
+		data, err = convertRoomAvatar(data)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	_, err = wa.Client.SetGroupPhoto(portalJID, data)
+	if err != nil {
+		return false, err
+	}
+
+	msg.Portal.AvatarMXC = msg.Content.URL
+	msg.Portal.AvatarHash = sha256.Sum256(data)
+	msg.Portal.AvatarSet = true
+
+	return true, nil
+}
+
+const avatarMaxSize = 720
+const avatarMinSize = 190
+
+func convertRoomAvatar(data []byte) ([]byte, error) {
+	src, _, err := image.Decode(bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode avatar: %w", err)
+	}
+
+	imageBounds := src.Bounds()
+	width, height := imageBounds.Max.X, imageBounds.Max.Y
+
+	isCorrectSize := width == height && avatarMinSize < width && width < avatarMaxSize
+	if isCorrectSize && http.DetectContentType(data) == "image/jpeg" {
```

Comment:
Instead of re-detecting content type here, you can just check if the second value returned by `image.Decode` is `"jpeg"`

```suggestion
	if isCorrectSize && decodedType == "jpeg" {
```